gift-certificate-application/service/src/test/java/com/epam/esm/service/impl/TagServiceImplTest.java

```diff
@@ -0,0 +1,133 @@
+    private static final BigDecimal PRICE = BigDecimal.valueOf(99.99);
+    private static final short DURATION = 100;
+    private static final Long GIFT_CERTIFICATE_ID = 1L;
+    private static final int PAGE = 1;
+    private static final int SIZE = 10;
+    private static final Long ENTITIES_NUMBER = 1L;
+    private Tag tag;
+    private TagDao tagDao;
+    private GiftCertificateDao giftCertificateDao;
+    private TagValidator tagValidator;
+    private PaginationValidator paginationValidator;
+    private TagServiceImpl tagService;
+    private List<Tag> tagList;
+    private List<GiftCertificate> giftCertificateList;
+    private CustomPagination pagination;
+    private List<String> tagNames;
+
+    @BeforeAll
+    void beforeAll() {
+        tag = new Tag();
+        tag.setId(ID);
+        tag.setName(NAME);
+        GiftCertificate giftCertificate = new GiftCertificate();
+        giftCertificate.setId(GIFT_CERTIFICATE_ID);
+        giftCertificate.setName(GIFT_CERTIFICATE_NAME);
+        giftCertificate.setDescription(DESCRIPTION);
+        giftCertificate.setPrice(PRICE);
+        giftCertificate.setDuration(DURATION);
+        tagDao = mock(TagDaoImpl.class);
+        giftCertificateDao = mock(GiftCertificateDao.class);
+        tagValidator = mock(TagValidatorImpl.class);
+        paginationValidator = mock(PaginationValidatorImpl.class);
+        tagService = new TagServiceImpl(tagDao, giftCertificateDao, tagValidator, paginationValidator);
+        tagList = new ArrayList<>();
+        tagList.add(tag);
+        giftCertificateList = new ArrayList<>();
+        giftCertificateList.add(giftCertificate);
+        pagination = new CustomPagination();
+        pagination.setPage(PAGE);
+        pagination.setSize(SIZE);
+        tagNames = tagList.stream().map(name -> tag.getName()).collect(Collectors.toList());
+    }
+
+    @Test
+    void testCreate() {
+        Tag expectedResult = tag;
+        when(tagValidator.isNameValid(any())).thenReturn(true);
+        when(tagDao.create(any())).thenReturn(expectedResult);
+        Tag actualResult = tagService.create(tag);
+        assertEquals(expectedResult, actualResult);
+    }
+
+    @Test
+    void testCreateShouldThrowException() {
+        when(tagValidator.isNameValid(any())).thenReturn(false);
+        assertThrows(FieldValidationException.class, () -> tagService.create(tag));
+    }
+
+    @Test
+    void testFindAll() {
+        List<Tag> expectedResult = tagList;
+        when(tagDao.findAll(pagination, Tag.class)).thenReturn(tagList);
+        when(tagDao.findEntitiesNumber(Tag.class)).thenReturn(ENTITIES_NUMBER);
+        when(paginationValidator.validatePagination(pagination, ENTITIES_NUMBER)).thenReturn(pagination);
+        List<Tag> actualResult = tagService.findAll(pagination);
+        assertEquals(expectedResult, actualResult);
+    }
+
+    @Test
+    void testFindById() {
+        Tag expectedResult = tag;
```

Comment:
I suppose you can directly use tag and do not create an additional variable

Reply:
It's done